src/Components/SingleArticle.jsx

```diff
@@ -24,6 +26,23 @@ export default function SingleArticle() {
     const clickShowComments = () => {
         setShowComments(true);
     }
+
+    const handleVote = (newVote) => {
+        patchVote(article_id, newVote)
+          .then(() => {
+            setSingleArticle((prevState) => ({
+              ...prevState,
+              votes: prevState.votes + newVote,
+            }));
+            setHasVoted(newVote);
```

Comment:
On slower connections I can spam click these buttons multiple times and send multiple patch requests through because hasVoted is being set after patchVote has succeeded (setHasVoted in the then block on line 37). To stop this from happening you want to setHasVoted straight away at the top of your handle vote function. If you want to se what I'm talking about, go into your browser dev console and throttle the network speed to simulate a slow 3g connection. See what happens right now when you click the button a few times 😱 Also - can we improve on the UX to account for delays like this in future?